lib/include/event-driven/vDrawSkin.h

```diff
@@ -0,0 +1,1159 @@
+
+private:
+
+
+int* getID(const int index)const {
+
+        static int result[3];
+        int triangleID,taxelInTriangle;
+        bool success =0;
+
+
+        for(auto mapit=repr2TaxelList.begin();mapit!=repr2TaxelList.end();mapit++){
+            auto listIntriangle= mapit->second;
+            auto it = std::find(listIntriangle.begin(), listIntriangle.end(), index);
+            if(it != listIntriangle.end()){
+                triangleID = mapit->first;
+                //taxelInTriangle = std :: distance(listIntriangle.begin(),it);
+                taxelInTriangle = index-16*(triangleID-3)/12;
+                success =1;
+                break;
+            }
+        }
+
+        if (triangleID==-1){
+            for(auto mapit=repr2TaxelList.begin();mapit!=repr2TaxelList.end();mapit++){
+            auto listIntriangle= mapit->second;
+            auto it = std::find(listIntriangle.begin(), listIntriangle.end(), index-1);
+            if(it != listIntriangle.end()){
+                triangleID = mapit->first;
+                break;
+                }
+            }
+            if(taxelInTriangle%2==0)taxelInTriangle=6;
+            else taxelInTriangle = 10;
+        }
+
+        triangleID = (triangleID-3)/12;
+        result[0] = triangleID;
+        result[1] = taxelInTriangle;
+        result[2] = success;
+        return result;
+    };
+
+
+ public :
+
+    Triangle_10pad(){
+
+        nTaxels=12;
+
+        xoffset = -200;
+        yoffset = 600;
+
+
+        dX = new double[nTaxels];
+        dY = new double[nTaxels];
+
+        dX[8]=-128.62; dY[8]=222.83;
+        dX[10]= 0; dY[10]=296.55;
+        dX[ 9]= 0; dY[ 9]=445.54;
+        dX[ 11]= 128.62; dY[ 11]=222.83;
+        dX[ 0]= 257.2; dY[ 0]=0;
+        dX[ 6]= -256.3; dY[ 6]=-147.64;
+        dX[ 1]=  385.83; dY[1]=-222.83;
+        dX[ 2]=128.62; dY[ 2]=-222.83;
+        dX[ 3]=0.0; dY[ 3]=0.0;
+        dX[ 4]=-128.62; dY[ 4]=-222.83;
+        dX[ 5]=-385.83; dY[ 5]=-222.83;
+        dX[ 7]=-257.2; dY[ 7]=0.0;
+    };
+
+    void initRepresentativeTaxels(std::vector<int> taxelMap){
+        //yarp::os::RecursiveLockGuard rlg(recursive_mutex);
+        std::list<int> mapp(taxelMap.begin(), taxelMap.end());
+        mapp.sort();
+        mapp.unique();
+        double s=0;
+
+        size_t mappsize = mapp.size();
+        for (size_t i = 0; i < mappsize; i++)
+        {
+            repr2TaxelList[mapp.front()]=vectorofIntEqualto(taxelMap,mapp.front());
+            mapp.pop_front();
+        }
+        //assign correct spacing between addresses: every 12 5 empty addresses
+        for (auto it =repr2TaxelList.begin();it!=repr2TaxelList.end();it++){
+            for (auto it2 =  it->second.begin(); it2 !=it->second.end(); it2++)
+            {
+                if(it->first==-1){
+                    s+=1;
+                    *it2 = 10000; //dummy value to avoid problems. The algorithm below works only for contiguous triangles in assigning a correct value
```

Comment:
Ugly and error prone

Reply:
yeah agreed, but there is some not easy to understand interpretation of the data we have. I don't want to play with this now.